spi.java.hints/src/org/netbeans/spi/java/hints/JavaFixUtilities.java

```diff
@@ -1394,7 +1394,18 @@ private void rewrite(Tree from, Tree to) {
         OPERATOR_PRIORITIES.put(Kind.UNSIGNED_RIGHT_SHIFT_ASSIGNMENT, 15);
         OPERATOR_PRIORITIES.put(Kind.XOR_ASSIGNMENT, 15);
     }
-    
+
+    /**
+     * Checks whether {@code tree} can be used in places where a Primary is
+     * required (for instance, as the receiver expression of a method invocation).
+     * @param tree the tree to check
+     * @return {@code true} iff {@code tree} can be used where a Primary is
+     * required.
+     */
+    public static boolean isPrimary(@NonNull Tree tree) {
```

Comment:
API is being changed, so I think that old good API change procedure should happen. See http://wiki.netbeans.org/API_Development; http://wiki.netbeans.org/VersioningPolicy
You need to bump up Specification version of the spi.java.hints, write relevant apichanges.xml entry. Add proper "since" javadoc tag to this added method.
You need to increase the required spi.java.hints spec version In module java.hints which uses this new API.

It's often better to separate API change(s) into their own PR, which should come first

Reply:
I created PR #320 for the API change.